src/controllers/bill.js

```diff
@@ -47,29 +48,213 @@ module.exports.billsOfAMemberInABusiness = async (req, res) => {
+        ).session(session);
+        await session.commitTransaction();
+
+        return res.send({ message: "transaction recorded", bill });
+      } else {
+        throw new Error("Transaction amount is greater than sub total");
+      }
+    } else {
+      let totalSum = 0;
+      for (let i = 0; i < billData.partialTransactions.length; i++) {
+        totalSum += billData.partialTransactions[i].amount;
+      }
+      let diff = billData.subtotal - totalSum;
+      if (totalSum < billData.subtotal && amount <= diff) {
+        let partialObj = {};
+        partialObj.amount = amount;
+        partialObj.reference = reference;
+        partialObj.method = PAYMENT_METHOD_MANUAL;
+        partialObj.transactionType = type;
+        partialObj.paidAt = now;
+        partialObj.updateMethod = PAYMENT_METHOD_MANUAL;
+        partialObj.processDate = now;
+        transactionArray = billData.partialTransactions;
+        transactionArray.push(partialObj);
+        let update = {
+          $set: {
+            partialTransactions: transactionArray,
+          },
+        };
+        let options = { new: true, useFindAndModify: false };
+
+        let bill = await Bill.findByIdAndUpdate(
+          billId,
+          update,
+          options
+        ).session(session);
+        await session.commitTransaction();
+
+        return res.send({ message: "transaction recorded", bill });
+      } else {
+        throw new Error("No due left cannot record this transaction");
+      }
+    }
+
+  } catch (err) {
+    await session.abortTransaction();
+    return res.status(422).send({ message: err.message });
+  } finally {
+    session.endSession();
+  }
+};
 
-    let bill = await Bill.findByIdAndUpdate(billId, update, options);
+module.exports.deleteTransactions = async (req, res) => {
+  const session = await mongoose.startSession();
+  session.startTransaction();
+  try {
+    let { billId, transactionId } = req.body;
+    let billData = await Bill.findById(billId);
+    if (billData) {
+      if (billData.partialTransactions) {
+        let { partialTransactions } = billData;
+        let newTransactions = partialTransactions.filter((transaction) => {
+          return transaction._id != transactionId;
+        });
+        let update = {
+          $set: {
+            partialTransactions: newTransactions,
+          },
+        };
+        let options = { new: true, useFindAndModify: false };
 
-    return res.send({ message: "transaction recorded", bill });
+        let bill = await Bill.findByIdAndUpdate(
+          billId,
+          update,
+          options
+        ).session(session);
+        await session.commitTransaction();
+        return res.send({ message: "transaction deleted", bill });
+      }
+    }
   } catch (err) {
+    await session.abortTransaction();
     return res.status(422).send({ message: err.message });
+  } finally {
+    session.endSession();
   }
 };
 
+module.exports.updateTransactions = async (req, res) => {
```

Comment:
can you break the code into functions (each function should have a single responsibility/functionality)?

Reply:
done